[MRG] adding right sidebar

choldgraf opened the pull request:
This adds an "on this page" sidebar to the right of each page with internal links. This does the following:

* Add `localtoc.html` content in a sidebar to the right
* Slight modifications to page layout to make sure there's room for this content

If the screen becomes too narrow, then it simply hides the localtoc content.

Here's a GIF of what this looks like for the Z2JH docs:

![2019-01-16_10-33-40](https://user-images.githubusercontent.com/1839645/51270596-36f15180-197a-11e9-89d2-9ab74bf17ca0.gif)

choldgraf committed:
adding right sidebar

choldgraf commented:
cc @jupyterhub/binder-team @jupyterhub/jupyterhubteam since this is relevant to all the jupyterhub-related documentation. What do folks think about this?

Zsailer commented:
👍 This makes me really happy. I find myself scrolling back-and-forth often.

consideRatio commented:
Yes this is excellent i think, i also find myself doing that @zsailer :D

willingc commented:
Nice! Thanks @choldgraf

betatim commented:
Excellent!

Is the local TOC "sticky" (as in it always stays visible when scrolling down the page)? I also find myself having to scroll up and down (and all around) too often.

choldgraf commented:
Indeed it is!

minrk commented:
ooh, nice!

choldgraf committed:
small css fixes

choldgraf commented:
Updated the CSS to improve the spacing a little bit, and also made the right sidebar disappear a little bit sooner to avoid it getting too cluttered on narrow screens. I updated the GIF to show current behavior.

I'm ready to merge if folks wanna give this a shot! Also if folks are OK w/ it, I'll cut a new release on PIP once it's merged so that we can get this into use in our docs!
